todo_app.js

```diff
@@ -1,5 +1,108 @@
 
-    var todosListEl;
+
+
+function TodoApp() {
+    //your code to run since DOM is loaded and ready;
+    this.initApp();
+}
+
+document.addEventListener("DOMContentLoaded", function() {
+    var todoApp = new TodoApp();
+});
+
+
+TodoApp.prototype.initApp = function() {
+    this._todosListEl = document.getElementById('todos'); // ������ �� ��� ������� ������ �������
+    var addButton = document.querySelector('div > input:last-child');
+    addButton.addEventListener('click', this.addTodo.bind(this));
+    TodoManager.init();
+    this.renderList();
+};
+
+
+TodoApp.prototype.checker = function (event) {
+    var index = event.target.getAttribute('data-index');
+    event.target.parentNode.parentNode.classList.toggle('completed');
+    TodoManager.toggleTodo(index);
+};
+
+
+TodoApp.prototype.removeTodo = function (event) {
+    var index = event.target.getAttribute('data-index');
+    TodoManager.removeTodo(index);
+    this.renderList();
+};
+
+
+TodoApp.prototype.renderList = function() {
+    // renderTodos(todosFromLocalStorage);
+    var todoList = TodoManager.loadTodos();
+    this._renderedTodoList = todoList.map(function(todo, index) {
+        return this.renderTodo(todo, index);
+    }, this);
+
+    this._todosListEl.innerHTML = this._renderedTodoList.join('');
+
+    this.handlersToCheckboxes();
+    this.handlersToButtons();
+};
+
+
+//bind handlers to checkboxes
+TodoApp.prototype.handlersToCheckboxes = function() {
+    var checkboxesList = this._todosListEl.querySelectorAll('input'); // ������ ���������
+    checkboxesList.forEach(function (item) {
+        item.addEventListener("change", this.checker);
+    }, this);
+};
+
+//bind handlers to buttons
+TodoApp.prototype.handlersToButtons = function() {
+    var buttonslist = this._todosListEl.getElementsByClassName('remover');
+    for (var i = 0; i < buttonslist.length; i++) {
+        buttonslist[i].addEventListener("click", this.removeTodo.bind(this));
+    }
+};
+
+
+TodoApp.prototype.addTodo = function() {
+    // save todo to LS
+    var todoText = document.getElementById('task').value;
+
+    // add todo to list and bind checkbox event
+    if (todoText === '') {
+        alert('Please write something');
+    } else {
+        TodoManager.addTodo({
+            text: todoText
+        });
+        this._todoString = this.renderTodo({text:todoText});
```

Comment:
Do you really need this and above calculated elements to be stored as properties?